### PR TITLE
Add support for X-Forwarded-Prefix

### PR DESCRIPTION
**Pull request by Miouge1**
This PR adds support for X-Forwarded-Prefix header that is added by some proxies when manipulating the path of requests (example https://docs.traefik.io/basics/#matchers).

This should allow to run Kuberos behind a proxy with a path manipulaiton for instance: example.com/kuberos/ with the callback example.com/kuberos/ui

Where currently hosting Kuberos at example.com/kuberos/ the callback passed to OIDC is example.com/ui

I'm open to any other ideas on how to tackle this.

### CODE REVIEW

File: kuberos.go
Comment (negz) [inline kuberos.go]: I'd prefer we only ranged over `r.Header` once. This could be a switch statement:

```go
for h, v := range r.Header {
  switch h {
    case elbHeaderForwardedProto:
      // Existing check for HTTPS protocol
    case headerForwardedPrefix:
      // Set the URL path
  }
}
```

File: kuberos.go
Comment (negz) [inline kuberos.go]: A nitpick: would you mind renaming `elbHeaderForwarded` to `headerELBForwarded` in the above two lines. That way all the header related constants will start with `header`.

**Review by negz — CHANGES_REQUESTED** [review]
@Miouge1 Thanks for the contribution! I've left two small comments.

Have you confirmed Kuberos works as expected in both the cases, i.e. when the new header is set and when it is not?

**Comment by Miouge1** [comment]
I did not test this, since I ran into problems building the code locally with statik.

**Comment by negz** [comment]
@Miouge1 Understood. We'll need to make sure these changes work as expected before they're merged. Perhaps I can help you with your build issues?

**Comment by Miouge1** [comment]
Thanks for the offer, but I figured it out on a fresh VM.

So, I used `curl -v -H 'X-Forwarded-Prefix: /foo/' localhost:10003` to test, I get a
```
HTTP/1.1 303 See Other
< Location: https://localhost:1234/oauth/authorize?access_type=offline&client_id=foobar&prompt=consent&redirect_uri=http%3A%2F%2Flocalhost%3A10003%2Ffoo%2Fui&response_type=code&scope=openid+email+profile&state=bar
```
Note the redirect_uri=http%3A%2F%2Flocalhost%3A10003%2F**foo**%2Fui as expected.

When testing with the UI, I notice a JS XHR 404 error `/kubecfg`, so I have fixed that in the latest commit.

File: kuberos.go
Comment (Miouge1) [reply kuberos.go]: OK, makes sense. I change that to a switch

File: kuberos.go
Comment (Miouge1) [reply kuberos.go]: I think it would be better to remove the `elb` part all together, since those headers are not AWS specific. So I converted that to `headerForwarded`

File: cmd/kuberos/kuberos.go
Comment (negz) [inline cmd/kuberos/kuberos.go]: Did this line sneak in from one of the other PRs?

**Review by negz — APPROVED** [review]

**Comment by negz** [comment]
@Miouge1 looks good to me, aside from the one comment. Happy to merge this if you're confident it works with and without the new header.

**Comment by amit-handa** [comment]
Hi,
I am using kuberos,dex via traefik ingress. all are on latest versions. I am unable to redirect.
the curl command specified in the comments above is working *correctly* though. 
sorry, if I shouldnt comment here, I ll create an issue otherwise.
logs of the invocation are as follows:


time="2018-05-22T09:19:13Z" level=debug msg="vulcand/oxy/roundrobin/rr: begin ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-IN,en-GB;q=0.9,en-US;q=0.8,en;q=0.7\"],\"Upgrade-Insecure-Requests\":[\"1\"],\"User-Agent\":[\"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3359.181 Safari/537.36\"],\**"X-Forwarded-Prefix\":[\"/kuberos\"]**},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"k8s.gemsol.in:31689\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"192.168.219.64:42094\",\"RequestURI\":\"/\",\"TLS\":null}"
...
...
time="2018-05-22T09:19:13Z" level=debug msg="vulcand/oxy/forward: begin ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"http\",\"Opaque\":\"\",\"User\":null,\"Host\":\"192.168.135.52:5556\",\"Path\":\"\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-IN,en-GB;q=0.9,en-US;q=0.8,en;q=0.7\"],\"Upgrade-Insecure-Requests\":[\"1\"],\"User-Agent\":[\"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3359.181 Safari/537.36\"],\"X-Forwarded-Prefix\":[\"/\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"k8s.gemsol.in:31689\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"192.168.219.64:42094\",\"RequestURI\":\"/dex/auth?client_id=gem-kube\\u0026prompt=consent\\u0026redirect_uri=**https%3A%2F%2Fk8s.gemsol.in%3A31689%2Fui**\\u0026response_type=code\\u0026scope=openid+offline_access+groups+email\\u0026state=9454ad698774a8d771efa17d8f36b85c314630154c01b6a4f46da708b654801e\",\"TLS\":null}"

**Pull request merged**
